### PR TITLE
docs (notifications): usage and style guidance for Callout

### LINKED ISSUE #3913: [Callout] website docs: component
## Acceptance criteria

Add `callout` as a variant to the [Notification component](https://carbondesignsystem.com/components/notification/usage). Highlighted in the list below and the place you will need to add content for static notification. 

Some additional tasks have been added to improve the notification usage docs in general. This will help with understanding these component variants and better align with our content template. If you see other places to enhance the understanding of these docs and still align to our [content template](https://carbondesignsystem.com/contributing/documentation/#usage-template:-for-components-with-multiple-variants) then feel free to add them in. 

Related PR for this effort [#15532](https://github.com/carbon-design-system/carbon/pull/15532)

#### Deliverables
- [x] **Content draft:** Write a first draft of the content required below. It can be written either in boxnote first or Figma (Jeannie has kindly made an editable version of the docs [here](https://www.figma.com/design/zJ3X6LfnvOGQDW8zyURrpw/Notification---Usage?node-id=0-1&t=6SHsscrYiGyXSb8W-1).) This draft is mostly focused on the written content. Images are not required yet. Once you have a first draft, please share it with the team in slack for review. 
- [x] **Draft with images:** Once you have edit to your written copy ready mock the page up in Figma  (if you haven't done so yet). This draft should include images. Share with the team in a meeting when ready for a review. 
- [x] **Pull request**: Open a PR to submit the content for a final review before publishing. 

----

# Usage tab

## Overview section

- [x] Add a paragraph under [overview](https://carbondesignsystem.com/components/notification/usage#overview) introducing notifications. You can reference the [notification pattern](https://carbondesignsystem.com/patterns/notification-pattern/) for content if needed. You can also rewrite the [When to use](https://carbondesignsystem.com/components/notification/usage#when-to-use) section if needed. 
- [x] Add `Callout` to the [variants table](https://carbondesignsystem.com/components/notification/usage#variants)

## Formatting section

`### Anatomy`
- [x] Add `Callout` to the existing image. Place below inline so its easy to compare
- [x] Add any annotations needed to the anatomy list. For example, define that close does not apply to callout. 

`### Status`
- [x] Create new `H3`section for notification status. Migrate the content from the notification pattern [notification status](https://carbondesignsystem.com/patterns/notification-pattern/#notification-status) section. It will live in both places. Content can be edited if need but should be the same on both pages. 

## Callout section
This is a new section and where the bulk of the documentation work will be. Determine where to add the static variant amongst the other 3 variant sections ([Inline](https://carbondesignsystem.com/components/notification/usage#inline-notifications), [Toast](https://carbondesignsystem.com/components/notification/usage#toast-notifications), [Actionable](https://carbondesignsystem.com/components/notification/usage#actionable-notifications)) 

`## Callout`

- [x] Write an introduction section that explains what a static notification is and how its different than the other components (focus especially on explaining how it is different than inline). Include examples for when to use this variant and how it addresses the use case / solves a problem for the user.

`### Callout formatting` 
- [x] Create a image with micro layouts showing where Callout commonly goes, refer to [inline](https://carbondesignsystem.com/components/notification/usage#inline-formatting) for an example. 

`#### Sizing`
- [x] List out any component sizes that are available and provide some guidance about how to choose the best size for the use case. Other topics that may be covered here include considerations around maximum heights, maximum widths, and responsive resizing.

`#### Placement` 
- [x] Discuss placement of the component in a broader context, that is: alongside other components, within the grid, etc. 

`### Behaviors`
- [x] Add any additional behaviors or features of static notification here. The other component types have [Dismissal](https://carbondesignsystem.com/components/notification/usage#dismissal) here, is a relevant topic here as well. You can change the heading title to be more relevant to the content if needed.

# Style tab

## Structure section
- [x] Add a H3 sections for Callout. Copy the [inline notification](https://carbondesignsystem.com/components/notification/style#inline-notification) and modify it to callout specifications

# Accessibility tab
Out of scope. This tab has not been updated to the new accessibility template yet. We will make a separate issue for this work.

### PR DESCRIPTION
**Pull request by benjamin-kurien**
Closes #3913

Usage & style document for Notification component has been updated to include guidance for a 4th variant named Callout.

#### Changelog

**New**

- Anchor link to Callout section
- Added a paragraph under overview introducing notifications.
- Added a section to describe various notification statuses.
- Added guidance to main elements section for changes related Callout
- Added guidance to the style tab for specifications related to the  Callout variant.

**Changed**

- Added Callout to the variants table.
- Added Callout to the anatomy image, improved the overall layout for the image, and changed the anatomy list to individually address each variant.
- Added documentation for callout including its overview, guidance for actions, formatting and placement, and the required images
- Heading for variants changed to be consistent.

**Removed**

- Content for main elements that were specific to any variant was removed from this section. This is instead explained in detail under the relevant sections.
- Recommendation to use only the title to communicate the entire message - was removed. Guidance to keep the message short and concise was retained.

### CODE REVIEW

File: src/pages/components/notification/usage.mdx [hunk 1]
```diff
@@ -90,15 +97,60 @@ although some product teams also support banners and notification centers.
...
+<Title> Deciding what to use </Title>
+
+| Status        | Usage                                                                                                                        | Color  | Icon                 |
+| ------------- | ---------------------------------------------------------------------------------------------------------------------------- | ------ | -------------------- |
+| Informational | Provide additional information to users that may not be tied to their current action or task                                 | Blue   | `information filled` |
+| Success\*     | Confirm a task was completed as expected                                                                                     | Green  | `checkmark filled`   |
+| Warning       | Inform users that they are taking actions that are not desirable or might have unexpected results                            | Yellow | `warning filled`     |
+| Error\*       | Inform users of an error or critical failure and optionally block the user from proceeding until the issue has been resolved | Red    | `error filled`       |
+
+\*[Callout](#callout) does not include success and error statuses.
```
[the remark this answers is not in the document]
Comment (aagonzales) [reply src/pages/components/notification/usage.mdx:153]:
```suggestion
<Caption>\*[Callout](#callout) does not include success and error statuses.</Caption>
```

File: src/pages/components/notification/usage.mdx [hunk 2]
```diff
@@ -301,23 +355,130 @@ are alternative routes to navigate to the link destination.
...
+![Example of callout placement](images/notification-usage-calloutplacement.png)
+
+<Caption>
+  Example of how a callout is placed when an inline error notification appears
+</Caption>
+
+</Column>
+</Row>
+
+#### Dismissal
```
[the remark this answers is not in the document]
Comment (benjamin-kurien) [reply src/pages/components/notification/usage.mdx:456]: Dismissal, Links and Actions should be H3. This is essential to differentiate them for the other h4 headings under the formatting image. I will make this correction.

**Comment by benjamin-kurien** [comment]
Hi @aagonzales I have updated the anatomy image as requested. Please share your review. Thank you.

**Review by aagonzales — APPROVED** [review]
Looks good to go! Good job!

File: src/pages/components/notification/usage.mdx [hunk 3]
```diff
@@ -301,23 +359,130 @@ are alternative routes to navigate to the link destination.
...
 
-All notifications, including actionable, persist by default until a user
-dismissed them. If using inline styling refer to
-[inline notifications](#inline-notifications) for inline dismissal. If using
-toast styling for an actionable notification, then the notification should
-remain on screen until the user dismisses it. With the notification remaining
-open, the user has enough time to interact with the button without the toast
-closing too soon. Refer to [toast notifications](#toast-notifications) for
-further information.
+Actionable notification, persist by default until a user dismiss them. If using
```
Comment (RichKummer) [inline src/pages/components/notification/usage.mdx:362]:
```suggestion
Actionable notifications persist by default until a user dismisses them. If using
```

File: src/pages/components/notification/usage.mdx [hunk 4]
```diff
@@ -301,23 +359,130 @@ are alternative routes to navigate to the link destination.
...
+
+If needed, actionable notifications can include a ghost button for inline
+notifications or a tertiary button for toasts. These action buttons enable users
+to address the notification or navigate to a page for more details.
+
+## Callout
+
+Callouts are used to highlight important information contextually within the
+contents of the page, and cannot be dismissed. Unlike other notification
+components they are not triggered by the user or system, rather load with the
```
Comment (RichKummer) [inline src/pages/components/notification/usage.mdx:385]:
```suggestion
components they are not triggered by the user or system, rather they load with the
```

File: src/pages/components/notification/usage.mdx [hunk 5]
```diff
@@ -301,23 +359,130 @@ are alternative routes to navigate to the link destination.
...
+Callouts are related to a particular page element and should be positioned near
+it to associate them more effectively.
+
+- **With button or input:** If important information affects an action, place
+  the callout above the relevant button or input. This highlighted message
+  ensures that users feel confident about their action before proceeding.
+- **With data or information:** Place the callout contextually, either before or
+  after the data being viewed by users, depending on what fits best.
+- **With inline notification:** When a callout and an inline notification need
+  to coexist during a part of the user journey, move the callout upward and
```
Comment (RichKummer) [inline src/pages/components/notification/usage.mdx:444]:
```suggestion
  to coexist during a part of the user journey, move the callout above the inline notification and
```

File: src/pages/components/notification/usage.mdx [hunk 6]
```diff
@@ -301,23 +359,130 @@ are alternative routes to navigate to the link destination.
...
+### Links
+
+If needed, callouts can contain links. Links on callout should be used only when
+it provides additional context or relevant information necessary for the current
+user journey. It should not be used to promote navigation to a separate user
+journey.
+
+### Actions
+
+Unlike actionable notifications, callouts do not have action button of their
```
Comment (RichKummer) [inline src/pages/components/notification/usage.mdx:475]:
```suggestion
Unlike actionable notifications, callouts do not contain an action button of their
```

**Review by RichKummer — COMMENTED** [review]
Made a few minor edits just for clarity. Otherwise looks good!

File: src/pages/components/notification/usage.mdx [hunk 7]
```diff
@@ -301,23 +359,130 @@ are alternative routes to navigate to the link destination.
...
-All notifications, including actionable, persist by default until a user
-dismissed them. If using inline styling refer to
-[inline notifications](#inline-notifications) for inline dismissal. If using
-toast styling for an actionable notification, then the notification should
-remain on screen until the user dismisses it. With the notification remaining
-open, the user has enough time to interact with the button without the toast
-closing too soon. Refer to [toast notifications](#toast-notifications) for
-further information.
+Actionable notification, persist by default until a user dismiss them. If using
+inline styling refer to [inline notifications](#inline-notifications) for inline
```
Comment (annawen1) [inline src/pages/components/notification/usage.mdx:363]:
```suggestion
inline styling, refer to [inline notifications](#inline-notifications) for inline
```

**Review by annawen1 — APPROVED** [review]
After Richie's comments, LGTM!

**Review by RichKummer — APPROVED** [review]
Thanks for making the updates @benjamin-kurien ! LGTM

**Comment by benjamin-kurien** [comment]
Waiting for devs to make the required name changes to storybook, before merging this PR.

**Pull request merged**